app/models/customer.rb

```diff
@@ -1,5 +1,20 @@
 class Customer < ApplicationRecord
 
   has_many :invoices
+  has_many :transactions, through: :invoices
 
-end 
+  def self.top_customers
+    joins(:transactions)
+    .where('result = ?', 1)
+    .group(:id)
+    .select("customers.*, count('transactions.result') as top_result")
```

Comment:
top_result isn't really descriptive of what this is. maybe use a variable name like transaction_count?